ledger/cow.go

```diff
@@ -285,3 +290,34 @@ func (cb *roundCowState) commitToParent() {
 func (cb *roundCowState) modifiedAccounts() []basics.Address {
 	return cb.mods.Accts.ModifiedAccounts()
 }
+
+// CalculateTotals calculates the totals given the changes in the StateDelta.
+// these changes allow the validator to validate that the totals still align with the
+// expected values. ( i.e. total amount of algos in the system should remain consistent )
+func (cb *roundCowState) CalculateTotals() error {
+	// this method applies only for the top level roundCowState
+	if cb.commitParent != nil {
+		return nil
+	}
+	totals := cb.prevTotals
+	var ot basics.OverflowTracker
+	totals.ApplyRewards(cb.mods.Hdr.RewardsLevel, &ot)
+
+	for i := 0; i < cb.mods.Accts.Len(); i++ {
+		accountAddr, updatedAccountData := cb.mods.Accts.GetByIdx(i)
+		previousAccountData, lookupError := cb.lookupParent.lookup(accountAddr)
```

Comment:
this would fetch the old account state from the cache in `roundCowBase` ( it won't get to the accounts update ).

Reply:
This sounds like like. Maybe we can eliminate accountsBase from account updates as well

Reply:
accountsBase is meant to cache account data across multiple rounds. The cowroundbase is designed to cache the data only on the boundaries of a single round. I think that when refactoring the account updates, we can consider refactoring the cache as well.